src/Features/Test/EditAndContinue/EditAndContinueWorkspaceServiceTests.cs

```diff
@@ -2785,6 +2781,187 @@ partial class C { int Y = 2; }
+                static void F()
+                {
+                    Console.WriteLine(1);
+                }
+            }
+            """;
+
+        var sourceB1 = """
+            using System;
+            
+            interface I
+            {
+            }
+            """;
+
+        var sourceA2 = """
+            using System;
+            class A
+            {
+                static void F()
+                {
+                    Console.WriteLine(2);
+                }
+            }
+            """;
+
+        var sourceB2 = """
+            using System;
+
+            interface I
+            {
+                void F() {}
+            }
+            """;
+
+        using var _ = CreateWorkspace(out var solution, out var service);
+
+        solution = solution
+            .AddTestProject("A", out var projectAId)
+                .AddTestDocument(sourceA1, "A.cs", out var documentAId).Project.Solution
+            .AddTestProject("B", out var projectBId)
+                .AddTestDocument(sourceB1, "B.cs", out var documentBId).Project.Solution;
+
+        EmitAndLoadLibraryToDebuggee(solution.GetRequiredDocument(documentAId));
+        EmitAndLoadLibraryToDebuggee(solution.GetRequiredDocument(documentBId));
+
+        var debuggingSession = await StartDebuggingSessionAsync(service, solution);
+
+        // change the source (valid edit in A and rude edit in B):
+        solution = solution
+            .WithDocumentText(documentAId, CreateText(sourceA2))
+            .WithDocumentText(documentBId, CreateText(sourceB2));
+
+        // Rude Edit reported:
+        var diagnostics = await service.GetDocumentDiagnosticsAsync(solution.GetRequiredDocument(documentBId), s_noActiveSpans, CancellationToken.None);
+        AssertEx.Equal(
+            ["ENC0023: " + string.Format(FeaturesResources.Adding_an_abstract_0_or_overriding_an_inherited_0_requires_restarting_the_application, FeaturesResources.method)],
+            diagnostics.Select(d => $"{d.Id}: {d.GetMessage()}"));
+
+        // validate solution update status and emit:
+        var (updates, emitDiagnostics) = await EmitSolutionUpdateAsync(debuggingSession, solution);
+        Assert.Empty(emitDiagnostics);
+        Assert.Equal(ModuleUpdateStatus.RestartRequired, updates.Status);
+
+        // TODO: https://github.com/dotnet/roslyn/issues/78244
+        // Should emit delta for the valid change
+
+        //// check emitted delta:
+        //var delta = updates.Updates.Single();
+        //Assert.Empty(delta.ActiveStatements);
+        //Assert.NotEmpty(delta.ILDelta);
+        //Assert.NotEmpty(delta.MetadataDelta);
+        //Assert.NotEmpty(delta.PdbDelta);
+        //Assert.Equal(6, delta.UpdatedMethods.Length);  // F, C.C(), D.D(), E.E(int), E.E(int, int), lambda
+        //AssertEx.SetEqual([0x02000002, 0x02000003, 0x02000004, 0x02000005], delta.UpdatedTypes, itemInspector: t => "0x" + t.ToString("X"));
+
+        //debuggingSession.DiscardSolutionUpdate();
```

Comment:
Is this commented code intentional or should it have been deleted?

Reply:
Intentional. Following up in next PR.